File: backend/app/schemas/graph.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel


class GraphNode(BaseModel):
    id: str
    label: str
    type: str  # client, compte, beneficiaire, alerte
    x: Optional[float] = None
    y: Optional[float] = None
    alert: Optional[bool] = False
    details: Optional[Dict[str, Any]] = None


class GraphEdge(BaseModel):
    from_node: str
    to_node: str
    label: Optional[str] = None
    strong: Optional[bool] = False


class FinancialGraphOut(BaseModel):
    client_id: str
    client_nom: str
    noeuds: List[GraphNode]
    liens: List[GraphEdge]
    total_flux_detectes: float

File: LAKANA-/frontend/src/models/graph.ts
export interface GraphNode {
  id: string
  label: string
  type: "client" | "compte" | "beneficiaire" | "alerte" | string
  x?: number
  y?: number
  alert?: boolean
  details?: Record<string, any>
}

export interface GraphEdge {
  from_node: string
  to_node: string
  label?: string
  strong?: boolean
}

export interface FinancialGraph {
  client_id: string
  client_nom: string
  noeuds: GraphNode[]
  liens: GraphEdge[]
  total_flux_detectes: number
}
